src/comments.test.ts

```diff
@@ -0,0 +1,52 @@
+import ts from "typescript";
+import { describe, expect, it, vitest } from "vitest";
+
+import { forEachComment } from "./comments";
+import { createNodeAndSourceFile } from "./test/utils";
+import { isTsVersionAtLeast } from "./utils";
+
+describe("forEachComment", () => {
+	it("does not call the callback when the source has no comments", () => {
+		const { node, sourceFile } = createNodeAndSourceFile("let value;");
+		const callback = vitest.fn();
+
+		forEachComment(node, callback, sourceFile);
+
+		expect(callback).not.toHaveBeenCalled();
+	});
+
+	if (isTsVersionAtLeast(4, 3)) {
```

Comment:
This fails in TS 4.2: https://github.com/JoshuaKGoldberg/ts-api-utils/actions/runs/4815452608/jobs/8574152647?pr=205

...which makes me motivated to drop support for TS 4.2. #206

Reply:
I'm all for using newer versions